smartsim/_core/_install/redisaiBuilder.py

```diff
@@ -0,0 +1,260 @@
+        main_build_path: pathlib.Path,
+        verbose: bool = False,
+        source: t.Union[str, pathlib.Path] = "https://github.com/RedisAI/RedisAI.git",
+        version: str = "v1.2.7",
+    ) -> None:
+
+        self.platform = platform
+        self.mlpackages = mlpackages
+        self.build_env = build_env
+        self.verbose = verbose
+        self.source = source
+        self.version = version
+        self._root_path = main_build_path / "RedisAI"
+
+        self.cleanup_build()
+
+    @property
+    def src_path(self) -> pathlib.Path:
+        return pathlib.Path(self._root_path / "src")
+
+    @property
+    def build_path(self) -> pathlib.Path:
+        return pathlib.Path(self._root_path / "build")
+
+    @property
+    def package_path(self) -> pathlib.Path:
+        return pathlib.Path(self._root_path / "package")
+
+    def cleanup_build(self) -> None:
+        """Removes all directories associated with the build"""
+        shutil.rmtree(self.src_path, ignore_errors=True)
+        shutil.rmtree(self.build_path, ignore_errors=True)
+        shutil.rmtree(self.package_path, ignore_errors=True)
+
+    @property
+    def is_built(self) -> bool:
+        """Determine whether RedisAI and backends were built
+
+        :return: True if all backends and RedisAI module are in
+                 the expected location
+        """
+        backend_dir = CONFIG.lib_path / "backends"
+        rai_exists = [
+            (backend_dir / f"redisai_{backend_name}").is_dir()
+            for backend_name in self.mlpackages
+        ]
+        rai_exists.append((CONFIG.lib_path / "redisai.so").is_file())
+        return all(rai_exists)
+
+    @property
+    def build_torch(self) -> bool:
+        """Whether to build torch backend
+
+        :return: True if torch backend should be built
+        """
+        return "libtorch" in self.mlpackages
+
+    @property
+    def build_tensorflow(self) -> bool:
+        """Whether to build tensorflow backend
+
+        :return: True if tensorflow backend should be built
+        """
+        return "libtensorflow" in self.mlpackages
+
+    @property
+    def build_onnxruntime(self) -> bool:
+        """Whether to build onnx backend
+
+        :return: True if onnx backend should be built
+        """
+        return "onnxruntime" in self.mlpackages
+
+    def build(self) -> None:
+        """Build RedisAI
+
+        :param git_url: url from which to retrieve RedisAI
+        :param branch: branch to checkout
+        :param device: cpu or gpu
+        """
+
+        self.src_path.mkdir(parents=True)
+        self.build_path.mkdir(parents=True)
+        self.package_path.mkdir(parents=True)
+
+        retrieve(self.source, self.src_path, depth=1, branch=self.version)
+
+        self._prepare_packages()
+
+        for package in self.mlpackages.values():
+            self._patch_source_files(package.rai_patches)
+        cmake_command = self._rai_cmake_cmd()
+        build_command = self._rai_build_cmd
```

Comment:
just makin sure you didnt mean to call this as self._rai_build_cmd(). disregard if I am wrong

Reply:
Yes, `rai_build_cmd` is a property and not a method on this class. Might need to change this in the future to be more generic, but for now this seems to satisfy all the cases